rpc: cuda-specific ut changed to device-specific

dayanandav opened the pull request:
RPC CUDA-specific UT updated to be compatible with device-specific UT, including dependencies based on the current accelerator (e.g., CUDA/XPU).

Added RPC specific tests for XPU backend.

Fixes #ISSUE_NUMBER

dayanandav committed:
rpc cuda-specific ut changed to device-specific

RPC CUDA-specific UT updated to be compatible with
device-specific UT, including dependencies based on
the current accelerator (e.g., CUDA/XPU).

dayanandav committed:
rpc: add XPU device support and unit tests for Torch + TensorPipe

Introduce XPU-specific unit tests to cover RPC execution
paths when running with the Torch + TensorPipe transport.

cyyever commented on torch/testing/_internal/distributed/rpc/dist_autograd_test.py:
```diff
@@ -2523,7 +2524,7 @@ def test_gpu_simple(self):
     @skip_if_lt_x_gpu(1)
     @dist_init
```
Can we iterate "cuda" and "xpu" in the decoration?

dayanandav replied:
As of today RPC device UT were running only for cuda and xpu, so my suggestion is not to mention devices in decoration.

cyyever commented on torch/testing/_internal/distributed/rpc/common_utils.py:
```diff
@@ -0,0 +1,9 @@
+# mypy: allow-untyped-defs
```
Don't suppress mypy in new code

dayanandav committed:
Update common_utils.py

Review comment fix

dayanandav committed:
Update common_utils.py

lint error fix

dayanandav committed:
Merge branch 'pytorch:main' into develop

dayanandav committed:
Update test_tensorpipe_agent.py

Lint error fix

dayanandav committed:
Update common_utils.py

_get_device_name return default "cpu" as device type when accelerator fail to get the current_accelerator device.

dayanandav committed:
Update common_utils.py

Review comment fix
